feat(sindri): enable benchmarking for zk proof circuits over sindri infrastructure

aWN4Y25pa2EK opened the pull request:
feat(main.rs): add command line arguments for benchmark mode and number of proofs to improve configurability
feat(Cargo.toml): introduce optional features for benchmarking and metrics to allow modular builds
fix(sindri.json): change proving scheme from groth16 to compressed for improved efficiency in proof generation

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added a ZK proof benchmarking command with configurable runs via environment variables; prints configuration and results.
  - Enabled benchmark mode for proofs to surface cycle/count data and richer logs.
  - CLI gains benchmarking: concurrent proof tasks, scalable transaction batches, predictive timing, throughput metrics, and a comprehensive results table, with improved observability and error handling.

- Chores
  - Introduced shared metrics dependency and optional feature flags in scripts.
  - Removed a temporary dependency override in the API to simplify version management.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

aWN4Y25pa2EK committed:
feat(proof.rs): enable benchmark mode for cycle counts in proof generation to enhance performance analysis
feat(main.rs): add command line arguments for benchmark mode and number of proofs to improve configurability
feat(Cargo.toml): introduce optional features for benchmarking and metrics to allow modular builds
fix(sindri.json): change proving scheme from groth16 to compressed for improved efficiency in proof generation

aWN4Y25pa2EK committed:
chore: remove outdated architecture and context augmenter documentation to streamline project structure
feat: add benchmarking commands to Makefile for ZK proof performance testing
feat: integrate metrics library for enhanced performance monitoring and analysis
fix: update proof generation logic to include benchmark metadata for better observability
refactor: improve transaction handling in benchmark scenarios for accurate scaling predictions
docs: enhance comments and documentation for clarity on benchmarking and performance metrics

aWN4Y25pa2EK committed:
chore(vscode): add settings for Rust Analyzer and TOML formatting to enhance development experience

maschad commented on sindri.json:
```diff
@@ -2,7 +2,7 @@
     "name": "arda-credit",
     "circuitType": "sp1",
-    "provingScheme": "groth16",
+    "provingScheme": "compressed",
```
I think we want to use `groth16` as that's what we submit to Ethereum.

aWN4Y25pa2EK replied:
Thanks for flagging @maschad to provide some context the change to `compressed` was requested by @MSevey so please if we can confirm which of the 2 to use or we wait for today's meeting to decide ?

MSevey replied:
I added in my makefile comment that we can set this when running benchmarks, but we shouldn't change it on main.

maschad reviewed (changes requested):
Great work on this @aWN4Y25pa2EK  🚀 

LGTM minus that one minor comment

maschad committed:
Merge branch 'main' into clean-benchmarking-optimizations

MSevey commented on Makefile:
```diff
@@ -181,6 +181,83 @@ cover:
+	@echo "  • 100 concurrent 1000tx proofs: ~49K txs/second"
+	@echo "  • Platform saturation: 100+ concurrent proofs show degradation"
+	@echo "  • Optimal configuration: 10-20 concurrent proofs for consistent latency"
+
```
This can all be condensed into a single call. 
```suggestion
# =============================================================================
# ZK Proof Benchmarking Commands (requires SINDRI_API_KEY)
# =============================================================================

NUM_PROOFS=1
NUM_TRANSACTIONS=100

## bench-zk: Run a zk benchmark test. Usage: make bench-zk NUM_PROOFS=50 NUM_TRANSACTIONS=500
.PHONY: bench-zk
bench-zk
	@echo ""
	@echo "Prerequisites:"
	@echo "  • SINDRI_API_KEY set in .env"
	@echo "  • Update sindri.json provingScheme to compressed"
	@echo "  • Circuit deployed: make deploy"
	@echo ""
	@TOTAL_TXS=$$(($(NUM_PROOFS) * $(NUM_TRANSACTIONS))); \
	echo "🎛️  Running benchmark test..."; \
	echo "Configuration: $(NUM_PROOFS) proofs × $(NUM_TRANSACTIONS) transactions = $$TOTAL_TXS total transactions"; \
	echo ""
	NUM_PROOFS=$(NUM_PROOFS) NUM_TRANSACTIONS=$(NUM_TRANSACTIONS) cargo run --release --package arda-credit --features sindri,benchmark
```

MSevey commented on script/src/bin/main.rs:
````diff
@@ -483,3 +509,689 @@ fn create_test_loan_scenario() -> (LoanState, LoanBatch, [u8; 32], [u8; 32]) {
+/// - Optimal_C ≈ 300 concurrent proofs  
+/// - Queue_Penalty ≈ 0.3 seconds per additional proof
+/// ```
+///
````
This doesn't need to be captured here since it would be captured in our SLA with them. 
Not something we need to capture. We tell them what we need and they give it to us.

MSevey reviewed (changes requested):
let's revert the sindri.json and clean up the makefile

aWN4Y25pa2EK committed:
chore(Makefile): remove outdated benchmark commands to simplify the Makefile
feat(Makefile): add new benchmark command for zk tests with configurable parameters
refactor(main.rs): remove extensive comments on benchmarking formulas to clean up code
fix(sindri.json): change provingScheme from compressed to groth16 for compatibility

maschad reviewed (approved):
LGTM
